crates/proto/src/rr/domain/label.rs

```diff
@@ -224,14 +224,28 @@ impl Borrow<[u8]> for Label {
     }
 }
 
-fn is_safe_ascii(c: char, is_first: bool, for_encoding: bool) -> bool {
+fn is_valid_ascii(c: char, is_first: bool, for_encoding: bool) -> bool {
```

Comment:
Does it make sense to keep the `is_first` special casing if we've reached the conclusion the `_` character is allowed in non-hostnames without any requirement on position in the label?

Reply:
It's a good point, I can't think of a reason it should be retained...